docs/madr/decisions/041-meshservice-kubernetes-ux.md

```diff
@@ -0,0 +1,64 @@
+# `MeshService` Kubernetes UX
+
+* Status: accepted
+
+Technical Story: #9704
+
+## Context and Problem Statement
+
+The new `MeshService` resource in Kubernetes zones maps 1-1 to Kubernetes
+`Service` resources. Additionally, users should not have to manage
+`MeshServices` directly but instead a `MeshService` should be derived directly
+from its `Service` with the possibility of users annotating the `Service`
+to control specifics.
+
+## Considered Options
+
+* CP manages `MeshService` from `Service`
+* Users have to create `MeshService`
+
+## Decision Outcome
+
+We will convert `Services` to `MeshServices` in a Kubernetes controller,
+assuming:
+
+1. The namespace is part of the mesh with `kuma.io/sidecar-injection: "true"`
+1. The `Mesh` of the `Service` or its namespace exists.
+1. The `Service` hasn't opted out via `kuma.io/ignore: "true"`
+
+### Metadata
+
+The `ownerReference` of the `MeshService` points to the `Service`.
+Labels are copied from the `Service` object to the `MeshService` object.
```

Comment:
What happens with conflicts?

Reply:
Conflicts where?

Reply:
Should we also copy annotations?

Reply:
I'd avoid this for now if we don't have to do this

Reply:
No I meant if by some random reason there's already a MeshService with this name

Reply:
IMO it's the same answer to the same question but for `Dataplanes`

Reply:
I'm fine with the error. Is this an event on the service or the meshService?

Reply:
Also do we merge the labels even if the ownerReference don't match?

Reply:
On the Service I would say, the user likely doesn't know that the MeshService exists and would first look at the Service.

Reply:
> Also do we merge the labels even if the ownerReference don't match?

Is this a different case than the MeshService existing? How exactly would you get to this state otherwise? If the MeshService exists, we don't overwrite it and we don't touch the labels.

Reply:
No I think are right